core/src/main/java/tc/oc/pgm/rotation/VotingPool.java

```diff
@@ -119,6 +119,24 @@ public boolean addCustomVoteMap(MapInfo info) {
     return false;
   }
 
+  public Set<MapInfo> fillCustomMapVote() {
+    Set<MapInfo> selected = Sets.newHashSet();
+    if (customVoteMaps.size() < MAX_VOTE_OPTIONS) {
+      int needed = MAX_VOTE_OPTIONS - customVoteMaps.size();
+      int maxRetrys = MAX_VOTE_OPTIONS;
+      while (selected.size() < needed && maxRetrys > 0) {
+        MapInfo random = getRandom();
+        if (!isCustomMapSelected(random)) {
+          selected.add(random);
+          addCustomVoteMap(random);
+        } else {
+          maxRetrys--;
+        }
+      }
+    }
+    return selected;
```

Comment:
This is adding maps randomly, it'd be nice if it added maps based on the vote-selection logic of scores and weights

Reply:
That’s a really good idea. I’ll look into switching this logic around later today 👍

Reply:
Can we just auto-fill for the user? Why require them to fill?

Reply:
It's just an issue of how vote modifying commands were approached, it was discussed in #551 . It would be expected that if you just add a map to the vote it would "force" one map and keep the other 4 random. Instead, what vote managing commands do, is expect you to select 2-5 options for the user to pick from. Some form of a "setnext between this options".

I think vote commands should always replace options of whatever vote will be held next, and maybe optionally make it so it makes the vote *only* the maps added via command, but i heavily doubt that will ever be used (wich is what we currently have)

Reply:
I converted this PR to a draft and will work on adjusting the logic to handle vote management as you suggested @Pablete1234 👍